test/dtpools/src/dtpools_attr.c

```diff
@@ -58,7 +58,7 @@ static int construct_contig(DTP_pool_s dtp, int attr_tree_depth, DTPI_Attr_s * a
 
         count /= attr->u.contig.blklen;
 
-        if (VALUE_FITS_IN_UINTPTR_T(extent * count))
+        if (VALUE_FITS_IN_INTPTR_T(extent * count))
```

Comment:
Following up on your reply (https://github.com/pmodels/yaksa/pull/130/commits/5c4c84e0ebdf7636cb26293dbd211307aa87a535#r519047604), both `extent` and `count` is signed integer now, right?

Reply:
`extent` is `int64_t`.  This macro is checking if it fits in `intptr_t`.

Reply:
Does `INTPTR_T_MAX * 2` fit inside `intptr_t`? If `INTPTR_T_MAX * 2` overflow to negative, the macro will retrun `true`, right?

Reply:
Oh, I don't know what that `* 2` is for.  It should simply compare `val`.

Reply:
`If INTPTR_T_MAX * count overflow to negative`

Why is this happening?  It's `int64_t`, not `intptr_t`.

Reply:
What if `int64_t` and `intptr_t` is both `long int`?

Reply:
Then this:

> val is int64_t. So, the assumption in the code is that int64_t is large enough to accommodate it. The problematic type would be intptr_t which can be 32 bits.

Reply:
I see.